### PR TITLE
TDL-14950 removed the line that caused the bug and added a unittest

### PR DESCRIPTION
**Pull request by namrata270998**
# Description of change

- removed the line in the build_metadata_metadata function that caused the bug and added a unit test case

# Manual QA steps
 - Check if all the custom fields are successfully added in the catalog.json for metadata_visitor and metadata_accounts stream
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_pendo/discover.py [hunk 1]
```diff
@@ -63,16 +63,17 @@ def get_schema_property_type(schema_type):
     elif schema_type == '':
         return {}
 
-    raise Exception("No case matching JSON schema for property type: {}".format(schema_type))
+    raise Exception(
```
Comment (dbshah1212) [inline tap_pendo/discover.py:66]: Why it is showing changes in this file? If nothing changed then do not commit it.

Comment (dbshah1212) [inline tap_pendo/discover.py:66]: Use f-string instead of .format, it's an older practice.

File: tests/unittests/test_custom_fields.py [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+                        }
+                    }
+                }
+            }
+            }
+        }
+        mdata = {}
+        schema = {'properties': {}}
+        build_metadata_metadata(mdata, schema, custom_account_fields)
+        # LOGGER.info(f'>>>>>>>>>>>>>>>> {schema}')
```
Comment (umangagrawal-crest) [inline tests/unittests/test_custom_fields.py:191]: Remove commented code

File: tests/unittests/test_custom_fields.py [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+                        }
+                        }
+                    }
+                }
+            }
+        }
+        mdata = {}
+        schema = {'properties': {}}
+        build_metadata_metadata(mdata, schema, custom_visitor_fields)
+        self.assertEqual(schema, exoected_schema)
```
Comment (umangagrawal-crest) [inline tests/unittests/test_custom_fields.py:295]: Please correct the spelling of expected_schema

File: tests/unittests/test_custom_fields.py [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+                "dirty": True,
+                "is_hidden": False,
+                "is_deleted": False,
+                "is_calculated": False,
+                "is_per_app": False,
+                "never_index": False
+            }
+        }
+
+        exoected_schema = {
```
Comment (umangagrawal-crest) [inline tests/unittests/test_custom_fields.py:210]: Please correct the spelling of expected_schema

**Review by dbshah1212 — APPROVED** [review]

File: tap_pendo/discover.py [hunk 5]
```diff
@@ -71,8 +71,8 @@ def build_metadata_metadata(mdata, schema, custom_fields):
         schema['properties']['custom'] = {}
         schema['properties']['custom']['type'] = ["null", "object"]
         schema['properties']['custom']['additional_properties'] = "false"
-    for key, _ in custom_fields.items():
         schema['properties']['custom']['properties'] = {}
+    for key, _ in custom_fields.items():
         schema['properties']['custom']['properties'][key] = {}
```
Comment (karanpanchal-crest) [inline tap_pendo/discover.py:76]: @namrata270998 What change is done as part of this line?

File: tests/unittests/test_custom_fields.py [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+                "element_format": "",
+                "dirty": True,
+                "is_hidden": False,
+                "is_deleted": False,
+                "is_calculated": False,
+                "is_per_app": False,
+                "never_index": False
+            }
+        }
+        expected_schema = {
```
Comment (karanpanchal-crest) [inline tests/unittests/test_custom_fields.py:37]: @namrata270998 Please add detailed comments as part of this file

**Pull request closed without merging**